### PR TITLE
fix(freki): propagate room renames to training_samples labels

### LINKED ISSUE #57: fix(freki): room rename never updates training_samples.label — silently orphans ML training data
**Priority: P0** · Part of the 2026-06 deep code-quality review (#54)

## Problem

`update_room` (`freki/src/freki/routers/rooms.py:108-115`) updates `labels.room` (FK cascade) and explicitly updates `csi_samples.label`, but **never touches `training_samples.label`** — the denormalized label column (`mimir/src/csi_models/sql/002_training_samples.sql:31`) on the table Nornir actually trains from (`freki/src/freki/routers/training_data.py:103-120`).

After a rename:

- `GET /api/training-data?rooms=&lt;new_name&gt;` silently returns **zero** pre-rename rows
- the old-name rows are orphaned forever (no room row matches them)
- models trained after the rename silently lose all historical data for that room

## Fix

- In the same transaction as the rename, also run `UPDATE training_samples SET label = :new WHERE label = :old`, using the same `_set_backfill_timeouts` lock/statement-timeout pattern as the labels backfill (it touches an indexed column on a large table)
- Longer term, consider deriving `training_samples.label` via FK to `rooms` so renames can't desynchronize it
- Add a DB-backed regression test (none of `rooms.py` is currently tested — see the test-coverage sub-issue under #56)

### PR DESCRIPTION
**Pull request by dannysauer**
Fixes #57 (P0 from the 2026-06 review, epic #54). **Stack position: 1/5** — this PR is the base of a stacked series; later PRs target this branch and will retarget to `main` as each one merges.

## What was wrong

`PATCH /api/rooms/{name}` updated `labels.room` (FK cascade) and `csi_samples.label`, but never `training_samples.label` — the denormalized column `GET /api/training-data` filters on and Nornir trains from. After a rename, the renamed room's historical training data was silently orphaned under the old name.

## The fix

- The rename now rewrites `training_samples.label` in the same transaction, under a savepoint so a real failure aborts the whole rename rather than splitting the tables between two names.
- Installs where `training_samples` is owned by a different role (#34/#37) skip the sync with a `training_samples.rename_skipped` warning, matching the labels router's existing degraded mode.
- New `freki/tests/test_rooms.py` covers the rename rewrite, the permission-denied carve-out, and the floor-only path (this router previously had zero tests).
- `docs/api-reference.md` and the router docstring now state the full propagation.

Note: this PR keeps the rename UPDATEs inline; PR 2 in the stack (#62) adds the lock/statement timeouts and background-task pattern so they can't hang on compressed chunks.

`pytest freki/tests` — 30 passed. Pre-commit hooks pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_011P84UZViqZqR1dZpsD5CsC

---
_Generated by [Claude Code](https://claude.ai/code/session_011P84UZViqZqR1dZpsD5CsC)_

### CODE REVIEW

File: freki/src/freki/routers/rooms.py [hunk 1]
```diff
@@ -62,6 +67,34 @@ def name_not_empty(cls, v: str | None) -> str | None:
...
+    Runs inside the caller's transaction under a savepoint, so a real
+    failure aborts the whole rename instead of leaving the tables split
+    between two names. Installs where training_samples is owned by a
+    different role (#34/#37) skip the sync with a warning so the rename
+    itself still succeeds.
+    """
+    try:
+        async with session.begin_nested():
+            await session.execute(
+                text("UPDATE training_samples SET label = :new_name WHERE label = :old_name"),
```
[the remark this answers is not in the document]
Comment (dannysauer) [reply freki/src/freki/routers/rooms.py:84]: Agreed — this is a real concern and it's handled by the next PR in the stack: #96 moves the whole rename propagation (csi_samples + training_samples) off the request path into a background task under the shared `SET LOCAL lock_timeout`/`statement_timeout` guard, the same pattern the label-create path uses. This PR is kept minimal on purpose: it fixes the data-orphaning bug (#57) inline, and #96 (issue #62) fixes the hang class for both this path and `delete_label`. Merging the stack in order lands the bounded version.

---
_Generated by [Claude Code](https://claude.ai/code)_
